disnake/flags.py

```diff
@@ -514,12 +514,25 @@ class Intents(BaseFlags):
     value: :class:`int`
         The raw value. You should query flags via the properties
         rather than using this raw value.
+
+        .. versionchanged:: 2.6
+
+            This can be now be provided on initialisation.
     """
 
     __slots__ = ()
 
-    def __init__(self, **kwargs: bool):
-        self.value = self.DEFAULT_VALUE
+    def __init__(self, value: int = None, **kwargs: bool):
```

Comment:
```suggestion
    def __init__(self, value: Optional[int] = None, **kwargs: bool):
```

Reply:
this was intentional as users shouldn't be implementing providing None as the value.

Reply:
We'll eventually enable the `strictParameterNoneValue` pyright rule, at which point we'll have to change this anyway. Type checkers will already infer the parameter type as `Optional[int]` regardless of whether it's explicitly specified or just typed as `int`.